.examples/docker-compose/with-nginx-proxy-self-signed-ssl-collabora-office-mariadb/docker-compose.yml

```diff
@@ -0,0 +1,109 @@
+    container_name: nextcloud-proxy
+    networks:
+      - nextcloud_network
+    ports:
+      - 80:80
+      - 443:443
+    volumes:
+      - ./proxy/conf.d:/etc/nginx/conf.d:rw
+      - ./proxy/vhost.d:/etc/nginx/vhost.d:rw
+      - ./proxy/html:/usr/share/nginx/html:rw
+      - ./proxy/certs:/etc/nginx/certs:ro
+      - /etc/localtime:/etc/localtime:ro
+      - /var/run/docker.sock:/tmp/docker.sock:ro
+    restart: unless-stopped
+  
+  letsencrypt:
+    image: jrcs/letsencrypt-nginx-proxy-companion
+    container_name: nextcloud-letsencrypt
+    depends_on:
+      - proxy
+    networks:
+      - nextcloud_network
+    volumes:
+      - ./proxy/certs:/etc/nginx/certs:rw
+      - ./proxy/vhost.d:/etc/nginx/vhost.d:rw
+      - ./proxy/html:/usr/share/nginx/html:rw
+      - /etc/localtime:/etc/localtime:ro
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+    restart: unless-stopped
+  db:
+    image: mariadb
+    container_name: nextcloud-mariadb
+    networks:
+      - nextcloud_network
+    volumes:
+      - db:/var/lib/mysql
+      - /etc/localtime:/etc/localtime:ro
+    environment:
+    # Create a root password for the maraiadb instance.
+      - MYSQL_ROOT_PASSWORD=CREATE-A-SECURE-ROOT-PASSWORD-HERE
+    # Create a password for the nextcloud users.  If you have to manually connect your database you would use the nextcloud user and this password.
+      - MYSQL_PASSWORD=CREATE-A-SECURE-NEXTCLOUD-USER-PASSWORD-HERE
+      - MYSQL_DATABASE=nextcloud
+      - MYSQL_USER=nextcloud
+    restart: unless-stopped
+  
+  app:
+    image: nextcloud:latest
+    container_name: nextcloud-app
+    networks:
+      - nextcloud_network
+    depends_on:
+      - letsencrypt
+      - proxy
+      - db
+    volumes:
+      - nextcloud:/var/www/html
+      - ./app/config:/var/www/html/config
+      - ./app/custom_apps:/var/www/html/custom_apps
+      - ./app/data:/var/www/html/data
+      - ./app/themes:/var/www/html/themes
+      - /etc/localtime:/etc/localtime:ro
+    environment:
+    # The VIRTUAL_HOST and LETSENCRYPT_HOST should use the same publically reachable domain for your nextlcloud instance.
+      - VIRTUAL_HOST=cloud.DOMAIN.TLD
+      - LETSENCRYPT_HOST=cloud.DOMAIN.TLD
+    # This needs to be a real email as it will be used by let's encrypt for your cert and is used to warn you about renewals.
+      - LETSENCRYPT_EMAIL=YOUR-EMAIL@DOMAIN.TDL
+    restart: unless-stopped
+  collab:
+    image: collabora/code
+    container_name: nextcloud-collab
+    networks:
+      - nextcloud_network
+    depends_on:
+      - proxy
+      - letsencrypt
+    cap_add:
+     - MKNOD
+    ports:
+      - 9980:9980
+    environment:
+    # This nees to be the same as what you set your app domain too (ex: cloud.domain.tld).
+      - domain=cloud\\.DOMAIN\\.TDL
+      - username=admin
+    # Create a passoword for the collabora office admin page.
+      - password=CREATE-A-SECURE-PASSWORD-HERE
+      - VIRTUAL_PROTO=https
+      - VIRTUAL_PORT=443
```

Comment:
In the [`nginx-proxy` docs](https://github.com/nginx-proxy/nginx-proxy#multiple-ports) it reads:

> If your container exposes multiple ports, nginx-proxy will default to the service running on port 80. If you need to specify a different port, you can set a VIRTUAL_PORT env var to select a different one. If your container only exposes one port and it has a VIRTUAL_HOST env var set, that port will be selected.

So I got this up and running setting *only* `VIRTUAL_HOST`. The container exposes 9980 per default, and `nginx-proxy` will pick up on that automatically. So we can stay DRY and not specify 9980 multiple times, or at all for that matter.

My setup is a bit different (reverse proxy is one docker-compose, Nextcloud instance another), but I can confirm this works:

```yaml
  collabora:
    image: collabora/code
    networks:
      # Just the default network of this docker-compose (no need to declare, exists per default):
      - default
      # Outside access through `external` reverse-proxy network:
      - proxy_default
    depends_on:
      # NOT the reverse proxy, but nginx webserver for nextcloud-fpm image:
      - web
    cap_add:
     - MKNOD
    environment:
      - username=admin
      - password=password
      # Domain the service should be accessed from:
      - domain=${VIRTUAL_HOST}
      #
      - VIRTUAL_HOST=${COLLABORA_VIRTUAL_HOST}
      - LETSENCRYPT_HOST=${COLLABORA_VIRTUAL_HOST}
      #
      # Extra parameters to Collabora, see also
      # https://www.collaboraoffice.com/code/nginx-reverse-proxy/:
      # SSL terminates at the proxy
      - extra_params=--o:ssl.enable=false --o:ssl.termination=true
```

Note no ports were specified at all. `nginx-proxy` has created this entry in `/etc/nginx/conf.d/default` automatically:

```plaintext
# <office.domain.tld>
upstream <office.domain.tld> {
                                # Cannot connect to network of this container
                                server 127.0.0.1 down;
                                ## Can be connected with "proxy_default" network
                        # nextcloud_collabora_1
                        server 192.168.16.5:9980;
}
```

where `nextcloud_collabora_1` is the automatic container name, generated from the above `collabora` service in a `docker-compose.yaml` in a `nextcloud` directory. I redacted my domain for `<office.domain.tld>`.

Very simple this way!

Reply:
I confirm. Works great